pkg/features/kube_features.go

```diff
@@ -957,6 +957,9 @@ const (
 func init() {
 	runtime.Must(utilfeature.DefaultMutableFeatureGate.Add(defaultKubernetesFeatureGates))
 	runtime.Must(clientgofeaturegate.AddFeaturesToExistingFeatureGates(utilfeature.DefaultMutableFeatureGate))
+
+	// use our authoritative copy to replace client-go features
+	clientgofeaturegate.SetFeatureGates(utilfeature.DefaultMutableFeatureGate)
```

Comment:
@deads2k @benluddy 

First, I’ve just realised that we have many features. Some are defined in `kubernetes/pkg/features`, and others in `apiserver/pkg/features`. For `KCM`, features are registered from both packages. However, I'm not sure if this applies to all components. It depends on the imports :(

Secondly, the current implementation (component-base) does not allow adding a feature with the same name but a different scope (I doubt we would like to change it), and this is exactly my case. Because my feature will be disabled by default in client-go, but in the `apiserver/pkg/features` package, it will be enabled.

Thus, we cannot add

```
runtime.Must(clientgofeaturegate.AddFeaturesToExistingFeatureGates(utilfeature.DefaultMutableFeatureGate))
clientgofeaturegate.SetFeatureGates(utilfeature.DefaultMutableFeatureGate)
```


to `pkg/features/kube_features.go` because it will blow up for many (if not all) components.

I think what we could do is to add `clientgofeaturegate.SetFeatureGates(utilfeature.DefaultMutableFeatureGate)` to `cmd/kube-controller-manager/app/options/options.go` before `AddFlag`.

This would assume that features from `apiserver/pkg/features` are registered for `KCM` (true today) and that feature names must match across packages and that we would like to honor default value set by KCM not client-go pkg. Does it make sense ?

Reply:
Alternativaly we require different names for client-go based FGs but then we will leak FG names from `apiserver/pkg/features` and the other package that are useless basically. And we would have to decide somehow whether we would like to overwrite a client-go FG or not.

Reply:
different names for client-go and everything else.